File: src/utils/linear-scaler.js

class LinearScaler {

    constructor(inputRange, outputRange) {
        this._inputRange = inputRange;
        this._outputRange = outputRange;
    }

    scale(value) {
        var xMin = this._outputRange[0];
        var xMax = this._outputRange[1];
        var yMin = this._inputRange[0];
        var yMax = this._inputRange[1];
        var percent = (value - yMin) / (yMax - yMin);
        var outputX = percent * (xMax - xMin) + xMin;
        return outputX;
    }

    inverse(value) {
        var xMin = this._inputRange[0];
        var xMax = this._inputRange[1];
        var yMin = this._outputRange[0];
        var yMax = this._outputRange[1];
        var percent = (value - yMin) / (yMax - yMin);
        var outputX = percent * (xMax - xMin) + xMin;
        return outputX;
    }

}

export default LinearScaler;

File: src/prediction-python/linear_scaler.py


class LinearScaler:
    def __init__(self, inputRange, outputRange):
        self._inputRange = inputRange
        self._outputRange = outputRange

    def scale(self, value):
        xMin = self._outputRange[0];
        xMax = self._outputRange[1];
        yMin = self._inputRange[0];
        yMax = self._inputRange[1];
        percent = (value - yMin) / (yMax - yMin);
        outputX = percent * (xMax - xMin) + xMin;
        return outputX;

    def inverse(self, value):
        xMin = self._inputRange[0];
        xMax = self._inputRange[1];
        yMin = self._outputRange[0];
        yMax = self._outputRange[1];
        percent = (value - yMin) / (yMax - yMin);
        outputX = percent * (xMax - xMin) + xMin;
        return outputX;
